test/fixtures/dialects/vertica/array.sql

```diff
@@ -11,14 +11,15 @@ CREATE TABLE sal_emp (
 
 SELECT ARRAY[[1, 2], [3, 4]];
 
-SELECT ARRAY[row(1, 2), row(1, 3)];
-
-SELECT
-    name,
-    num,
-    gpa
-FROM students
-WHERE major = ARRAY[row('Science', 'Physics')];
+-- Need to add support for complex datatypes
```

Comment:
This seems like a regression?

Reply:
Yes, that's true, but if you try to use row datatype not inside array it will fail because it's not implemented, so it seems better to have more specific array support and later add a row datatype

Current changes in the array segment allow me to parse sections with Jinja like:
```
select
    array[
    {% for metric_event in funnel_events %}
        {% for window_size in window_sizes %}
            null,
        {% endfor %}
    {% endfor %}
    null
]::array[varchar] as agg_data
from tab;
```

Without new changes, it will violate rule RF06 Missing quoted keyword identifier varchar